sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -580,9 +582,14 @@ object RmmRapidsRetryIterator extends Logging {
             lastException = ex
 
             if (!topLevelIsRetry && !causedByRetry) {
-              // we want to throw early here, since we got an exception
-              // we were not prepared to handle
-              throw lastException
+              // If the exception is the result of a CUDF column size overflow, attempt split-retry.
+              ex match {
+                case _: CudfColumnSizeOverflowException => doSplit = true
```

Comment:
In order to prevent infinite loops, not that it is true today, should we have a limit for doSplit due to the size overflow? 

Perhaps something large like 100 retries. We don't expect that many retries to happen, but it would at least give us a way to exit in case we are stuck with some sort of bug causing a loop.

Reply:
Never mind. The function doing the splitting will have a natural "out" (1 row, cannot split 1 row). I think we are fine doing this without any limits on the plugin.

Reply:
>> something large like 100 retries.
> The function doing the splitting will have a natural "out"

I was thinking that as well. 100 retries might suggest 2**100 rows to start with, which is well more than the `cudf::size_type` range, anyway.